src/internal/connector/exchange/service_restore.go

```diff
@@ -221,20 +255,24 @@ func SendMailToBackStore(
 		return errors.New("message not Sent: blocked by server")
 	}
 
-	if len(attached) > 0 {
-		id := *sentMessage.GetId()
-		for _, attachment := range attached {
-			err := uploadAttachment(ctx, service, user, destination, id, attachment)
-			if err != nil {
-				errs = support.WrapAndAppend(
-					fmt.Sprintf("uploading attachment for message %s: %s",
-						id, support.ConnectorStackErrorTrace(err)),
-					err,
-					errs,
-				)
-
-				break
-			}
+	id := *sentMessage.GetId()
+
+	for _, attachment := range attached {
+		uploader := &mailAttachmentUploader{
+			userID:   user,
+			folderID: destination,
+			itemID:   id,
+			service:  service,
+		}
```

Comment:
same here, should be safe to pull this out of the loop and reuse a single instance

Reply:
Resolved here [a234ee4](https://github.com/alcionai/corso/pull/1345/commits/a234ee42cdfeaf1fa619a1546ffa946fd93b642a)